### ISSUE TITLE
clang cannot detect any of the lifetime issue here but GCC can

### ISSUE THREAD

**Issue by trcrsired** [llvm/llvm-project#87229]
https://learn.microsoft.com/en-us/cpp/sanitizers/error-stack-use-after-scope?view=msvc-170
```cpp
int *gp;
bool b = true;
int main() {
    if (b) {
        int x[5];
        gp = x+1;
    }
    return *gp;  // Boom!
}
```
clang (no warnings):
https://godbolt.org/z/qab3nP38f
gcc:
https://godbolt.org/z/Porf9Mejj

**Comment by steakhal** [comment]
This is detected by the Static Analyzer. There is no clang builtin diagnostic for this, it seems.
Just pass the `--analyze` to the clang invocation to see that in action.
https://godbolt.org/z/vahEY4sdo

Can I close this?
